app/models/main.py

```diff
@@ -1832,6 +1848,307 @@ def serialize(self):
         }
 
 
+class ProcessOutcome(db.Model):
+    __tablename__ = "process_outcomes"
+
+    RESULT_CHOICES = (
+        "awarded",
+        "cancelled",
+        "none-suitable",
+    )
+
+    id = db.Column(db.Integer, primary_key=True)
+    external_id = db.Column(db.BigInteger, nullable=False, default=random_positive_external_id, unique=True)
+    completed_at = db.Column(db.DateTime, nullable=True)
+
+    start_date = db.Column(db.Date, nullable=True)
+    end_date = db.Column(db.Date, nullable=True)
+    awarding_organisation_name = db.Column(db.String, nullable=True)
+    # should be enough scale to represent up to £1b - 1p
+    award_value = db.Column(db.Numeric(11, 2), nullable=True)
+
+    result = db.Column(db.String, nullable=False)
+
+    direct_award_project_id = db.Column(
+        db.Integer,
+        db.ForeignKey('direct_award_projects.id'),
+        nullable=True,
+    )
+    direct_award_search_id = db.Column(
+        db.Integer,
+        db.ForeignKey('direct_award_searches.id'),
+        nullable=True,
+    )
+    direct_award_archived_service_id = db.Column(
+        db.Integer,
+        db.ForeignKey('archived_services.id'),
+        nullable=True,
+    )
+
+    # NOTE though ProcessOutcome currently has the *ability* to be associated with a Brief/BriefResponse, at time of
+    # writing this is not yet *used* for reporting brief awards - those are still using their own mechanism, but should
+    # be ported over to this mechanism as soon as time allows.
+    brief_id = db.Column(
```

Comment:
We are foreign keying to brief and brief response here when we can access brief through brief response. This kind of denormalisation is premature at this point and is likely to introduce more problems than it solves (misalignments). If we really must access brief on this model we can add a property. 

Also the point about it not being in use yet.

Reply:
Re the key - I think this is so that we can enforce here that there is only one non-draft outcome for each brief - if we only linked through brief_id then we wouldn't be able to do that.

So much comes down to the decision to insist on implementing all of these constraints in the database. Quite heated debate has happened about that already, there are compelling arguments on both sides - but given that decision, I think this is logical.

Re not being in use yet - I've cleared this with @louzoid-gds - I'm not massively happy either, but given there's some complexity to the design, and @risicle has now implemented it, after some debate/heartache I don't want to rip this out.

Reply:
@benvand there is no misalignment possible here - that is the precise reason for it existing. The only reason for any denormalization existing in this model is for the ability it gives us to form constraints.